### DISCUSSION TITLE
How to choose more than one object using MRTK Color Picker

### DISCUSSION THREAD

**Discussion by Nyri-N** [microsoft/MixedRealityToolkit-Unity#10845] [Q&A]
Hello, I was wondering how to set the MRTK Color Picker up to affect multiple meshes (to affect either child objects or just multiple meshes). We're beginners in C# so apologies if this is a dumb question 😅

**Comment by RogPodge** [comment]
Unfortunately the color picker in it's current experiemental state is only designed to work in the example scene against things designated as "TargetObject (Mesh)" and "TargetObject (Sprite)"

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/6e061451d7caed1fcb7c324baf92be293efda4cf/Assets/MRTK/SDK/Experimental/ColorPicker/ColorPicker.cs#L211-L219

You can modify the color picker to work against an arbitrary number of objects by adjusting the code to store a list of objects that can be affected by it. These objects can have the proper color applied to them during the `ApplyColor()` method.

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/6e061451d7caed1fcb7c324baf92be293efda4cf/Assets/MRTK/SDK/Experimental/ColorPicker/ColorPicker.cs#L301-L334

Unforunately, I don't think the MRTK team has the capacity to tackle this feature request right now, but we'll gladly accept any community contribution to improve this component! It's still in the experimental state, so we are more likely to accept any breaking changes to it!
